ext/github.py

```diff
@@ -0,0 +1,170 @@
+            embed.description += f"**Owner:** {data['owner']['login']}\n"
+            embed.description += f"**Description:** {data['description'][0:50]}\n"
+            if len(data["description"]) > 50:
+                embed.description += "..."
+            else:
+                pass
+            embed.url = data["html_url"]
+        
+        await ctx.send(embed=embed)
+        
+    @github.command(name="search", aliases=["s"])
+    async def search_repository(self, ctx, repo_name: str):
+        """
+        This command will get the top 5 repositories sorted by stars.
+
+        Syntax- c!github s <repo name>
+        """
+        embed = await self.search_repos(repo_name, 5)
+        await ctx.send(embed=embed)
+        
+    async def search_repos(self, repo_name, no_of_repos: int):
+        embed = discord.Embed(colour = discord.Colour.blue())
+        url = "https://api.github.com/search/repositories"
+        params = {
+            "q": repo_name,
+            "sort": "stars"
+        }
+        session = self.bot.http_session
+        data = await self.fetch(session, url, params)
+        if data["total_count"] == 0:
+            embed.title = "Repository not found!"
+            return embed
+        embed.url = f"https://github.com/search?q={repo_name}"
+        i = 0
+        for x in data["items"]:
+            if no_of_repos == 5:
+                # For search command.
+                embed.title = f"Search results for {repo_name}"
+                field_name = f"Repo: {x['owner']['login']}/{x['name']}"
+            else:
+                # For get repository command.
+                embed.title = f"Name: {repo_name}"
+                field_name = f"Owner: {x['owner']['login']}"
+                embed.url = x["html_url"]
+            i+=1
+            if x["description"] is None:
+                message = "No description avilable."
+            else:
+                message = x["description"][0:50]
+
+            value_field = f"**:star:{x['stargazers_count']}/:fork_and_knife:{x['forks']}\nDescription :** {message}...\nLink : {x['html_url']}\n"
+            if no_of_repos != 1:
+                if i != no_of_repos:
+                    value_field += "."
+                else:
+                    pass
+            embed.add_field(
+                name = field_name,
+                value = value_field,
+                inline=False
+            )
+            if i == no_of_repos:
+                break
+        return embed
+
+    @github.command(name="issue", aliases=["i"])
+    async def get_issues(self, ctx, owner: str, repo: str, issue_id: int):
+        """
+        A command to get complete information about an issue.
+
+        Syntax- c!github i <owner of repo> <repo name> <issue id>
+        
+        """
+        session = self.bot.http_session
+        url = f"https://api.github.com/repos/{owner}/{repo}/issues/{issue_id}"
+        data = await self.fetch(session, url)
+        if data["message"]:
```

Comment:
If there is a message, why not showing that instead?

Reply:
the message just says "Not found" if the search failed.